lambda/schema.ts

```diff
@@ -11,19 +12,9 @@ export const CONTENT_TYPES = {
   URL_ENCODED: 'application/x-www-form-urlencoded'
 } as const;
 
-export const headersSchema = z
-  .object({
-    'content-type': z.string().optional(),
-    'Content-Type': z.string().optional()
-  })
-  .refine(
-    headers => {
-      return headers['content-type'] || headers['Content-Type'];
-    },
-    {
-      message: 'Missing Content-Type header'
-    }
-  );
+export const headersSchema = z.object({
+  'content-type': z.nativeEnum(CONTENT_TYPES)
+}).transform(obj => mapKeys(obj, (_, key) => key.toLowerCase()));
```

Comment:
This effectively calls `.toLowerCase()` on all the header keys prior to validating against the schema